File: apps/backend/app/schemas/firmware.py
from datetime import datetime

from pydantic import BaseModel, Field


class FirmwareCreate(BaseModel):
    version: str = Field(..., max_length=20)
    description: str | None = Field(default=None)
    released_at: datetime | None = None


class FirmwareUpdate(BaseModel):
    version: str | None = Field(default=None, max_length=20)
    description: str | None = None
    released_at: datetime | None = None
    active: bool | None = None


class FirmwareResponse(BaseModel):
    id: int
    uuid: str
    version: str
    description: str | None
    released_at: datetime | None
    active: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class DeviceFirmwareStatus(BaseModel):
    device_id: int
    name: str
    current: str
    latest: str
    status: str
    progress: int
    date: str | None = None

File: apps/web/src/types/firmware.ts
export interface Firmware {
  id: number
  uuid: string
  version: string
  description: string | null
  released_at: string | null
  active: boolean
  created_at: string
}

export interface DeviceFirmwareStatus {
  device_id: number
  name: string
  current: string
  latest: string
  status: string
  progress: number
  date: string | null
}
